src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PETypeParameterSymbol.cs

```diff
@@ -138,80 +139,99 @@ public override AssemblySymbol ContainingAssembly
-            PENamedTypeSymbol containingType;
-
-            if (_containingSymbol.Kind == SymbolKind.Method)
+            if (_lazyDeclaredConstraintTypes.IsDefault)
             {
-                containingMethod = (PEMethodSymbol)_containingSymbol;
-                containingType = (PENamedTypeSymbol)containingMethod.ContainingSymbol;
-            }
-            else
-            {
-                containingType = (PENamedTypeSymbol)_containingSymbol;
-            }
+                ImmutableArray<TypeSymbol> declaredConstraintTypes;
 
-            var moduleSymbol = containingType.ContainingPEModule;
-            var metadataReader = moduleSymbol.Module.MetadataReader;
-            GenericParameterConstraintHandleCollection constraints;
+                PEMethodSymbol containingMethod = null;
+                PENamedTypeSymbol containingType;
 
-            try
-            {
-                constraints = metadataReader.GetGenericParameter(_handle).GetConstraints();
-            }
-            catch (BadImageFormatException)
-            {
-                constraints = default(GenericParameterConstraintHandleCollection);
-                Interlocked.CompareExchange(ref _lazyBoundsErrorInfo, new CSDiagnosticInfo(ErrorCode.ERR_BindToBogus, this), CSDiagnosticInfo.EmptyErrorInfo);
-            }
+                if (_containingSymbol.Kind == SymbolKind.Method)
+                {
+                    containingMethod = (PEMethodSymbol)_containingSymbol;
+                    containingType = (PENamedTypeSymbol)containingMethod.ContainingSymbol;
+                }
+                else
+                {
+                    containingType = (PENamedTypeSymbol)_containingSymbol;
+                }
 
-            if (constraints.Count > 0)
-            {
-                var symbolsBuilder = ArrayBuilder<TypeSymbol>.GetInstance();
-                MetadataDecoder tokenDecoder;
+                var moduleSymbol = containingType.ContainingPEModule;
+                var metadataReader = moduleSymbol.Module.MetadataReader;
+                GenericParameterConstraintHandleCollection constraints;
 
-                if ((object)containingMethod != null)
+                try
                 {
-                    tokenDecoder = new MetadataDecoder(moduleSymbol, containingMethod);
+                    constraints = metadataReader.GetGenericParameter(_handle).GetConstraints();
                 }
-                else
+                catch (BadImageFormatException)
                 {
-                    tokenDecoder = new MetadataDecoder(moduleSymbol, containingType);
+                    constraints = default(GenericParameterConstraintHandleCollection);
+                    Interlocked.CompareExchange(ref _lazyConstraintsUseSiteErrorInfo, new CSDiagnosticInfo(ErrorCode.ERR_BindToBogus, this), CSDiagnosticInfo.EmptyErrorInfo);
                 }
 
-                foreach (var constraintHandle in constraints)
+                this.GetAttributes();
```

Comment:
>this.GetAttributes(); [](start = 16, length = 21)

Why are we binding attributes here? We have PEModile.HasIsUnmanagedAttribute helper we should use it instead